ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,153 @@
+import { Checkbox, FormControlLabel, ListItem } from "@mui/material";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import { CriteriaConfig, CriteriaPlugin, registerCriteriaPlugin } from "cohort";
+import produce from "immer";
+import React from "react";
+import * as tanagra from "tanagra-api";
+import { useUnderlay } from "../hooks";
+
+type Selection = {
+  id: number;
+  name: string;
+};
+
+interface Config extends CriteriaConfig {
+  attribute: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter() {
+    if (this.data.selected.length === 0) {
+      return null;
+    }
+
+    const operands = () =>
+      this.data.selected.map(({ id }) => ({
+        binaryFilter: {
+          attributeVariable: {
+            variable: "person",
+            name: this.data.attribute,
+          },
+          operator: tanagra.BinaryFilterOperator.Equals,
+          attributeValue: {
+            int64Val: id,
+          },
+        },
+      }));
+
+    return {
+      arrayFilter: {
+        operands: operands(),
+        operator: tanagra.ArrayFilterOperator.Or,
+      },
+    };
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const underlay = useUnderlay();
+  const attributeName = props.data.attribute;
+
+  const hintDisplayName = (hint: tanagra.EnumHintValue) =>
+    hint.displayName || "Unknown Value";
+
+  const enumHintValues = underlay.entities
+    .find((g) => g.name === underlay.primaryEntity)
+    ?.attributes?.find((attribute) => attribute.name === attributeName)
+    ?.attributeFilterHint?.enumHint?.enumHintValues;
+
+  const selectionIndex = (hint: tanagra.EnumHintValue) =>
+    props.data.selected.findIndex(
+      (row) => row.id === hint.attributeValue?.int64Val
+    );
+
+  if (enumHintValues?.length === 0) {
+    return <Typography>No attribute information!</Typography>;
+  } else {
```

Comment:
Since the `if` case always returns, there's no need to have this block in an `else` case and it adds an extra layer of nesting.

Reply:
Note!